### DISCUSSION TITLE
Unable to load web3js

### DISCUSSION THREAD

**Discussion by zj978643142** [quasarframework/quasar#9860] [General - Components / Directives / etc] [answered]
App •  ERROR  •  UI  in ./node_modules/eth-lib/lib/bytes.js

Module not found: Can't resolve imported dependency "crypto"
Did you forget to install it? You can run: npm install --save crypto

 App •  ERROR  •  UI  in ./node_modules/ethereumjs-util/dist/account.js

Module not found: Can't resolve imported dependency "assert"
Did you forget to install it? You can run: npm install --save assert

 App •  ERROR  •  UI  in ./node_modules/web3-eth-accounts/lib/index.js

Module not found: Can't resolve imported dependency "crypto"
Did you forget to install it? You can run: npm install --save crypto

 App •  ERROR  •  UI  in ./node_modules/web3-eth-accounts/node_modules/eth-lib/lib/bytes.js

Module not found: Can't resolve imported dependency "crypto"
Did you forget to install it? You can run: npm install --save crypto

 App •  ERROR  •  UI  in ./node_modules/web3-providers-http/lib/index.js

Module not found: Can't resolve imported dependency "http"
Did you forget to install it? You can run: npm install --save http

 App •  ERROR  •  UI  in ./node_modules/web3-providers-http/lib/index.js

Module not found: Can't resolve imported dependency "https"
Did you forget to install it? You can run: npm install --save https

 App •  ERROR  •  UI  in ./node_modules/xhr2-cookies/dist/xml-http-request.js

Module not found: Can't resolve imported dependency "http"
Did you forget to install it? You can run: npm install --save http

 App •  ERROR  •  UI  in ./node_modules/xhr2-cookies/dist/xml-http-request.js

Module not found: Can't resolve imported dependency "https"
Did you forget to install it? You can run: npm install --save https

If I copy a copy of Axios to modify it, I will be prompted with such an error
Even if I install it manually, I will still prompt this error

**Comment by MilosPaunovic** [comment]
Seems like none of these errors is related to Quasar, so you should probably try to get answer [here](https://github.com/ChainSafe/web3.js).

**Reply by zj978643142** [reply]
But when I use version 1 to introduce web3js in the way of Axios, there is no error

**Reply by MilosPaunovic** [reply]
What do you mean in the way of Axios, as a boot file? If so, could you paste its content here?

**Reply by metalsadman** [reply]
looks like something to do with webpack 5, maybe the lib is not compatible with it yet. https://stackoverflow.com/a/67703356

**Reply by zj978643142** [reply]
Thanks @MilosPaunovic @metalsadman

**Reply by arieltoledo** [reply]
@zj978643142 did you manage to fix it ? Im having the same error

**Comment by rstoenescu** [comment, marked as answer]
https://quasar.dev/start/upgrade-guide#nodejs-polyfills 
I've even written it as one of the examples, knowing `crypto` is highly used :)

<img width="844" alt="image" src="https://user-images.githubusercontent.com/4665641/159782466-ba5b008c-330d-4179-b5d5-20fcaebd8780.png">

**Reply by arieltoledo** [reply]
@rstoenescu you are the best !
